### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by turuslan**
### Description of the Change
cpp-libp2p tcp reconnect problems.
Api types.
Cbor buffering and stream.
Update actor state, check before creating account.
Compute unsealed sector cid.

### Benefits
Fixes

### Possible Drawbacks 
cpp-libp2p is not "master" but "dev", without hunter fork.

### CODE REVIEW

File: core/adt/channel.hpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ namespace fc::adt {
     using Queue = std::pair<std::vector<T>, bool>;
     using Handler = std::function<bool(boost::optional<T>)>;
     struct Closed {};
+    using Many = std::vector<std::shared_ptr<Channel<T>>>;
```
Comment (Alexey-N-Chernyshov) [inline core/adt/channel.hpp:26]: Is it used?

Comment (turuslan) [reply core/adt/channel.hpp:26]: Yes

File: core/api/api.hpp [hunk 2]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+
+  struct RetrievalOrder {
+    CID root;
+    uint64_t size;
+    TokenAmount total;
+    uint64_t interval;
+    uint64_t interval_inc;
+    Address client;
+    Address miner;
+    std::string peer;
```
Comment (Alexey-N-Chernyshov) [inline core/api/api.hpp:237]: Is it string or bytes? Change to PeerId?

File: core/api/api.hpp [hunk 3]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+
+  struct MarketBalance {
+    TokenAmount escrow, locked;
+  };
+
+  struct QueryOffer {
+    std::string error;
+    CID root;
+    uint64_t size;
+    TokenAmount min_price;
```
Comment (artyom-yurin) [inline core/api/api.hpp:217]:
```suggestion
    BigInt min_price;
```

File: core/api/api.hpp [hunk 4]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+  struct StartDealParams {
+    DataRef data;
+    Address wallet;
+    Address miner;
+    TokenAmount epoch_price;
+    uint64_t epochs;
+  };
+
+  struct MarketBalance {
+    TokenAmount escrow, locked;
```
Comment (artyom-yurin) [inline core/api/api.hpp:210]:
```suggestion
    BigInt escrow, locked;
```

File: core/api/api.hpp [hunk 5]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
     IpldObject state;
   };
 
+  struct StartDealParams {
+    DataRef data;
+    Address wallet;
+    Address miner;
+    TokenAmount epoch_price;
```
Comment (artyom-yurin) [inline core/api/api.hpp:205]:
```suggestion
    BigInt epoch_price;
```

File: core/api/api.hpp [hunk 6]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+    uint64_t epochs;
```
Comment (artyom-yurin) [inline core/api/api.hpp:206]: In lotus here are two more fields
MinBlockDuration and DealStartEpoch

As I understood `epochs` is start epoch
Can you add MinBlockDuration?

File: core/api/api.hpp [hunk 7]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+    TokenAmount min_price;
+    uint64_t payment_interval;
+    uint64_t payment_interval_increase;
+    Address miner;
+    std::string peer;
+  };
+
+  struct FileRef {
+    std::string path;
+    bool car;
```
Comment (artyom-yurin) [inline core/api/api.hpp:226]:
```suggestion
    bool is_car;
```

File: core/api/api.hpp [hunk 8]
```diff
@@ -195,6 +198,52 @@ namespace fc::api {
...
+  };
+
+  struct RetrievalOrder {
+    CID root;
+    uint64_t size;
+    TokenAmount total;
```
Comment (artyom-yurin) [inline core/api/api.hpp:232]:
```suggestion
    BigInt total;
```

File: core/api/api.hpp [hunk 9]
```diff
@@ -214,7 +263,25 @@ namespace fc::api {
...
+               Wait<None>,
+               const RetrievalOrder &,
+               const FileRef &)
+    API_METHOD(ClientStartDeal, Wait<CID>, const StartDealParams &)
+
+    API_METHOD(MarketEnsureAvailable,
+               Wait<None>,
+               const Address &,
+               const Address &,
+               TokenAmount)
```
Comment (artyom-yurin) [inline core/api/api.hpp:284]:
```suggestion
               BigInt)
```

File: core/api/api.hpp [hunk 10]
```diff
@@ -297,6 +365,7 @@ namespace fc::api {
 
     API_METHOD(Version, VersionResult)
 
+    API_METHOD(WalletBalance, TokenAmount, const Address &)
```
Comment (artyom-yurin) [inline core/api/api.hpp:368]:
```suggestion
    API_METHOD(WalletBalance, BigInt, const Address &)
```

**Pull request merged**